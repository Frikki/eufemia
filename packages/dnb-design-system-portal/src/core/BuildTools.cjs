const path = require('path')
const fs = require('fs-extra')

function shouldUsePrebuild() {
  const rootPath = path.dirname(require.resolve('@dnb/eufemia'))
  const buildPath = path.resolve(rootPath, 'build')
  const packageJsonPath = path.resolve(buildPath, 'package.json')
  const prebuildExists = fs.existsSync(packageJsonPath)

  return Boolean(prebuildExists && process.env.NODE_ENV === 'production')
}
module.exports.shouldUsePrebuild = shouldUsePrebuild
