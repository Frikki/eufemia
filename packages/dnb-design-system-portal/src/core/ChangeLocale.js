import React from 'react'
import { Dropdown } from 'dnb-ui-lib/src'

import Context from 'dnb-ui-lib/src/shared/Context'
import { getLang, setLang } from './portalProviders'

export default function ChangeLocale(props) {
  const { locale, setLocale } = React.useContext(Context)

  return (
    <Dropdown
      value={getLang(locale)}
      data={{ 'en-US': 'English', 'nb-NO': 'Norsk' }}
      on_change={({ data: { selected_key } }) => {
        setLocale(selected_key)
        setLang(selected_key)
      }}
      {...props}
    />
  )
}
