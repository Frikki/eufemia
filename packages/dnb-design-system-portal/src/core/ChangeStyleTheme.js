import React from 'react'
import { Dropdown } from '@dnb/eufemia/src'

import {
  getThemes,
  getTheme,
  setTheme,
} from 'gatsby-plugin-eufemia-theme-handler'

export default function ChangeStyleTheme(props) {
  const themes = getThemes()
  const themeName = getTheme()

  const date = Object.entries(themes).reduce((acc, [key, { name }]) => {
    acc[key] = capitalizeFirstLetter(name)
    return acc
  }, {})

  return (
    <Dropdown
      value={themeName}
      data={date}
      on_change={({ data: { value } }) => {
        setTheme(value)
      }}
      {...props}
    />
  )
}

function capitalizeFirstLetter(s) {
  return s.charAt(0).toUpperCase() + s.slice(1)
}
