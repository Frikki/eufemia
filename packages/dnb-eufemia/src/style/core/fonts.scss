/*
 * Typography
 *
 * The font-family properties are defined in the properties.scss file, like:
 * --font-family-default:
 *
 */

// Font usage
.dnb-typo-regular {
  font-weight: normal;
  font-family: var(--font-family-default);
  font-style: normal;
}
.dnb-typo-medium {
  font-weight: var(--font-weight-medium);
  font-family: var(--font-family-default);
  font-style: normal;
}
.dnb-typo-bold {
  font-weight: var(--font-weight-bold);
  font-family: var(--font-family-default);
  font-style: normal;
}
.dnb-typo-mono-regular {
  font-weight: normal;
  font-family: var(--font-family-monospace);
  font-style: normal;
}

$fonts-path: '../../assets/fonts' !default;

// Std and Regular
@font-face {
  font-weight: normal; // have to be the same as: --font-weight-regular
  font-family: DNB;
  font-style: normal;
  src: url('#{$fonts-path}/DNB-Regular.woff2') format('woff2'),
    url('#{$fonts-path}/DNB-Regular.woff') format('woff'),
    url('#{$fonts-path}/DNB-Regular.ttf') format('truetype');
  font-display: fallback;
}

// Medium
@font-face {
  font-weight: 500; // have to be the same as: --font-weight-medium
  font-family: DNB;
  font-style: normal;
  src: url('#{$fonts-path}/DNB-Medium.woff2') format('woff2'),
    url('#{$fonts-path}/DNB-Medium.woff') format('woff'),
    url('#{$fonts-path}/DNB-Medium.ttf') format('truetype');
  font-display: fallback;
}

// Bold
@font-face {
  font-weight: 600; // have to be the same as: --font-weight-bold
  font-family: DNB;
  font-style: normal;
  src: url('#{$fonts-path}/DNB-Bold.woff2') format('woff2'),
    url('#{$fonts-path}/DNB-Bold.woff') format('woff'),
    url('#{$fonts-path}/DNB-Bold.ttf') format('truetype');
  font-display: fallback;
}

// Mono-Regular
@font-face {
  font-weight: normal; // have to be the same as: --font-weight-regular
  font-family: DNBMono;
  font-style: normal;
  src: url('#{$fonts-path}/DNBMono-Regular.woff2') format('woff2'),
    url('#{$fonts-path}/DNBMono-Regular.woff') format('woff'),
    url('#{$fonts-path}/DNBMono-Regular.ttf') format('truetype');
  font-display: fallback;
}
