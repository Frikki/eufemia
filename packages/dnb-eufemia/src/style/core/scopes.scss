/*
 * Scopes
 *
 */

@import './reset.scss';

@mixin coreDefault() {
  color: var(--color-black-80, #333);
  font-weight: var(--font-weight-basis);
  font-size: var(--font-size-small); // has to be 16px
  font-family: var(--font-family-default);
  font-style: normal;
  line-height: var(--line-height-basis);
  word-break: break-word;

  /**
  * Ensure consistency and use the same as HTML reset -> html {...}
  * between base and code package
  */
  -moz-tab-size: 4;
  tab-size: 4;

  // The new DNB font needs font smoothing to be thinner on webkit and FF
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;

  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;

  @content;
}

@mixin bodyDefault() {
  @include reset();
  @include coreDefault();

  @media print {
    color: black;
    // Use hard coded colors, because of theming and color swaps
    background-color: white;
  }

  @content;
}

@mixin htmlDefault() {
  html {
    font-size: 100%;
    scroll-behavior: smooth;

    // IS_SAFARI_MOBILE
    @supports (-webkit-touch-callout: none) {
      @supports (font: -apple-system-body) {
        font: -apple-system-body; /* stylelint-disable-line */
      }
    }
  }

  // reset ePlatform css
  @include resetLegacyStyles();
}

@mixin componentReset() {
  @include coreDefault();
  @include coreReset();

  margin: 0;
  padding: 0;

  @content;
}
