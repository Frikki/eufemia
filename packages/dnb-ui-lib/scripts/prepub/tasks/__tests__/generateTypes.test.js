/**
 * Scripts test
 *
 */

import { transformFileAsync } from '@babel/core'
import nodePath from 'path'
import {
  createTypes,
  babelPluginDefaults,
  babelPluginPropTypesRelations,
  babelPluginCorrectTypes,
  babelPluginIncludeDocs
} from '../generateTypes'
import { fetchPropertiesFromDocs } from '../fetchPropertiesFromDocs'

describe('babelPluginPropTypesRelations', () => {
  it('has to match snapshot', async () => {
    const file = nodePath.resolve(
      __dirname,
      '__mocks__/PrimaryComponent.js'
    )
    const sourceDir = nodePath.dirname(file)

    const { code } = await transformFileAsync(file, {
      plugins: [
        ['@babel/plugin-proposal-class-properties', { loose: true }],
        [babelPluginPropTypesRelations, { sourceDir }]
      ],
      ...babelPluginDefaults
    })

    expect(code).toMatchSnapshot()
  })
})

describe('babelPluginCorrectTypes', () => {
  it('has to match snapshot', async () => {
    const file = nodePath.resolve(
      __dirname,
      '__mocks__/PrimaryComponent.js'
    )
    const sourceDir = nodePath.dirname(file)

    const { code } = await transformFileAsync(file, {
      plugins: [
        ['@babel/plugin-proposal-class-properties', { loose: true }],
        [babelPluginPropTypesRelations, { sourceDir }],
        [
          babelPluginCorrectTypes,
          {
            /**
             * If strictMode is enabled,
             * it will transform "string + bool" or "string + number" in to string or bool only
             */
            strictMode: true
          }
        ]
      ],
      ...babelPluginDefaults
    })

    expect(code).toMatchSnapshot()
  })
})

describe('babelPluginIncludeDocs', () => {
  const docsDir = nodePath.resolve(__dirname, '__mocks__')
  const file = nodePath.resolve(docsDir, 'PrimaryComponent.js')
  let docs = {}

  it('has to match docs snapshot', async () => {
    docs = await fetchPropertiesFromDocs({
      file,
      docsDir,
      findFiles: ['PrimaryComponent.md']
    })
    expect(docs).toMatchSnapshot()
  })

  it('has to match code snapshot', async () => {
    const sourceDir = nodePath.dirname(file)

    const { code } = await transformFileAsync(file, {
      plugins: [
        ['@babel/plugin-proposal-class-properties', { loose: true }],
        [babelPluginPropTypesRelations, { sourceDir }],
        [
          babelPluginCorrectTypes,
          {
            /**
             * If strictMode is enabled,
             * it will transform "string + bool" or "string + number" in to string or bool only
             */
            strictMode: true
          }
        ],
        [
          babelPluginIncludeDocs,
          {
            docs
          }
        ]
      ],
      ...babelPluginDefaults
    })

    expect(code).toMatchSnapshot()
  })
})

describe('generateTypes', () => {
  it('has to match snapshot', async () => {
    const docsDir = nodePath.resolve(__dirname, '__mocks__')
    const file = nodePath.resolve(docsDir, 'PrimaryComponent.js')
    const [{ destFile, definitionContent }] = await createTypes([file], {
      isTest: true,
      docsDir,
      findFiles: ['PrimaryComponent.md']
    })

    expect(destFile).toContain('__mocks__/PrimaryComponent.d.ts')
    expect(definitionContent).toMatchSnapshot()
  })
})
