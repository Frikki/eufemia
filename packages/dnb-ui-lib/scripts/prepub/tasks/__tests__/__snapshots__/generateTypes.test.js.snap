// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTypes has to match snapshot 1`] = `
"import * as React from 'react';
export interface PrimaryComponentSpacing {
  top?: string | number | boolean;
  right?: string | number | boolean;
  bottom?: string | number | boolean;
  left?: string | number | boolean;
}
export type PrimaryComponentFoo = string | number | boolean;
export type PrimaryComponentTop = string | number | boolean;
export type PrimaryComponentRight = string | number | boolean;
export type PrimaryComponentBottom = string | number | boolean;
export type PrimaryComponentLeft = string | number | boolean;
export type PrimaryComponentSecondary = string | number | boolean;
export type PrimaryComponentSecondaryFoo = string | number | boolean;
export interface PrimaryComponentSecondarySpacing {
  secondary?: string | number | boolean;
}
export interface PrimaryComponentProps
  extends React.DOMAttributes<React.SyntheticEvent> {
  boolean?: boolean;
  number?: number;
  /**
   * Text \`spacing\` or \`children\`.
   */

  spacing?: PrimaryComponentSpacing;
  foo?: PrimaryComponentFoo;
  /**
   * Text \`margin-top\`.
   */

  top?: PrimaryComponentTop;
  /**
   * Text \`margin-right\`.
   */

  right?: PrimaryComponentRight;
  /**
   * Text \`margin-bottom\`.
   */

  bottom?: PrimaryComponentBottom;
  /**
   * Text \`margin-left\`.
   */

  left?: PrimaryComponentLeft;
  /**
   * Text \`secondary\`.
   */

  secondary?: PrimaryComponentSecondary;
  secondary_foo?: PrimaryComponentSecondaryFoo;
  secondary_spacing?: PrimaryComponentSecondarySpacing;
  /**
   * Text \`spacing\` or \`children\`.
   */

  children?: React.ReactNode;
}
export default class PrimaryComponent extends React.Component<
  PrimaryComponentProps,
  any
> {
  render(): JSX.Element;
}
"
`;
